utils/plottools.py

```diff
@@ -11,11 +11,24 @@ def __init__(self, exp, figpath, dims, units, ticks=None):
         self.DIMS      = dims
         self.DIM_UNITS = units
         self.DIM_TICKS = ticks or self._default_dim_ticks()
+        if 't' in self.DIMS.keys():
+            self.update_time_mapper()
+
+    def update_time_mapper(self):
+        self.TIME_MAPPER={'hour':[mpl.dates.HourLocator(interval=self.DIM_TICKS['t']),'%H'],\
+                          'minute':[mpl.dates.MinuteLocator(interval=self.DIM_TICKS['t']),'%H\n%M'],\
+                          }
```

Comment:
I think this attribute `TIME_MAPPER` can be a list, like `[mpl.date.xxxLocator, date_format]`. 
it is determined because the `self.DIM_UNITS['t']` is defined at initialization.
not dictionary ~~

Reply:
I was trying to create a general dict in the first place.
Then I realized the tick interval and the corresponding units should be set at the same time.
So you're right this should be a list!
Thank you for reviewing codes so carefully!